src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -190,6 +190,12 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
    */
   expanded: any = undefined;
 
+  /**
+   * To prevent making changes to the original columns list, that might change
+   * how the table is renderer a second time, we now clone that list into a
+   * local variable and only use the clone.
+   */
+  localColumns: CdTableColumn[];
```

Comment:
Hm as `this.columns` is only changed during `ngInit` I'm interested about the what component failed because of that change.

Reply:
Components didn't actually "fail", but since we add the details and check buttons to the left of the table each time the table is rendered, if we keep changing tabs it would keep adding more and more columns with those buttons.